providers/go-feature-flag/src/test/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProviderTest.java

```diff
@@ -180,41 +176,50 @@ public void initialize(EvaluationContext evaluationContext) throws Exception {
         assertThrows(ProviderNotReadyError.class, ()-> g.getBooleanEvaluation("bool_targeting_match", false, this.evaluationContext));
 
         String providerName = "shouldReturnNotReadyIfNotInitialized";
-        OpenFeatureAPI.getInstance().setProvider(providerName, g);
+        OpenFeatureAPI.getInstance().setProviderAndWait(providerName, g);
```

Comment:
test is _shouldReturnNotReadyIfNotInitialized_, with _setProvider_ is finishes faster and verifies it is not ready.